module/VuFindTheme/src/VuFindTheme/View/Helper/Slot.php

```diff
@@ -0,0 +1,259 @@
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFindTheme\View\Helper;
+
+/**
+ * Slot view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Slot extends \Zend\View\Helper\AbstractHelper
+{
+    /**
+     * End saving methods
+     *
+     * @const string
+     */
+    const SET   = 'SET';
+    const PREPEND = 'PREPEND';
+    const APPEND = 'APPEND';
+
+    /**
+     * Storage for strings to be concatinated to the front of a block
+     *
+     * @var array of arrays
+     */
+    protected $blockPrepends = [];
+
+    /**
+     * Storage for strings saved to slots
+     *
+     * @var array
+     */
+    protected $blocks = [];
+
+    /**
+     * Storage for strings to be concatinated to the end of a block
+     *
+     * @var array of arrays
+     */
+    protected $blockAppends = [];
+
+    /**
+     * Call stack to handle nested slots
+     *
+     * @var array
+     */
+    protected $stack = [];
+
+    /**
+     * Get the Slot instance. Create if instance doesn't exist.
+     *
+     * @param string $name  Name of target block for action
+     * @param any    $value Optional shortcut parameter to set a value
+     *
+     * @return Slot|string|any
+     */
+    public function __invoke($name, $value = null)
+    {
+        $this->stack[] = $name;
+        if ($value != null) {
+            return $this->set($value);
+        }
+        return $this;
+    }
+
+    /**
+     * Shortcut to get if no methods are called on invoke.
+     *
+     * @return string|any
+     */
+    public function __toString()
+    {
+        return $this->get();
+    }
+
+    /**
+     * Checks for content to provide isset functionality.
+     *
+     * @return boolean
+     */
+    public function isset()
+    {
+        $name = array_pop($this->stack);
+        return isset($this->blockPrepends[$name]) ||
+            isset($this->blocks[$name]) ||
+            isset($this->blockAppends[$name]);
+    }
+
+    /**
+     * Helper function to return blocks with prepends and appends.
+     * Prepends, blocks, and appends are separated byspacestopreventthisfromhappening
+     *
+     * Non-string data can be stored in a slot but prepend and append
+     * will cause it to be concatinated into a string.
+     *
+     * @param string $name Name of target block for action
+     *
+     * @return string|any
+     */
+    protected function build($name)
+    {
+        $pre = $this->blockPrepends[$name] ?? [];
+        $block = $this->blocks[$name] ?? '';
+        $post = $this->blockAppends[$name] ?? [];
+        if (!empty($pre) || !empty($post)) {
+            $ret = implode(' ', $pre) . ' ' . $block . ' ' . implode(' ', $post);
+            return trim($ret);
+        }
+        if (empty($block)) {
```

Comment:
I'm concerned that there may be some edge cases here that need to be more carefully addressed. For example, in records/home.phtml, you set a slot to `''`. My suspicion is that because that value is empty, the slot will display the default value instead of the blank string. It may be necessary to be a little more strict about differentiating between null and non-null values. It's definitely worth writing some test cases to demonstrate how we want this to behave.